cms/djangoapps/models/settings/course_metadata.py

```diff
@@ -260,7 +266,30 @@ def validate_and_update_from_json(cls, descriptor, jsondict, user, filter_tabs=T
                 'The proctoring provider cannot be modified after a course has started.'
                 ' Contact {support_email} for assistance'
             ).format(support_email=settings.PARTNER_SUPPORT_EMAIL or 'support')
-            errors.append({'message': message, 'model': filtered_dict.get('proctoring_provider')})
+            errors.append({'message': message, 'model': proctoring_provider_model})
+
+        # Require a valid escalation email if Proctortrack is chosen as the proctoring provider
```

Comment:
Can we wrap this logic about escalation email into a different function? Adding into this function makes this function difficult to read.